coral-spark/src/main/java/com/linkedin/coral/spark/transformers/FallBackToHiveUDFTransformer.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.spark.transformers;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlOperator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linkedin.coral.com.google.common.collect.ImmutableSet;
+import com.linkedin.coral.common.transformers.SqlCallTransformer;
+import com.linkedin.coral.hive.hive2rel.functions.VersionedSqlUserDefinedFunction;
+import com.linkedin.coral.spark.containers.SparkUDFInfo;
+import com.linkedin.coral.spark.exceptions.UnsupportedUDFException;
+
+
+/**
+ * After failing to transform UDF with {@link TransportableUDFTransformer},
+ * we use this transformer to fall back to the original Hive UDF defined in
+ * {@link com.linkedin.coral.hive.hive2rel.functions.StaticHiveFunctionRegistry}.
+ * This is reasonable since Spark understands and has ability to run Hive UDF.
+ * Check `CoralSparkTest#testFallBackToHiveUDFTransformer()` for an example.
+ */
+public class FallBackToHiveUDFTransformer extends SqlCallTransformer {
```

Comment:
General comment about the PR:

Coral-spark module should reflect the transformation of Coral IR -> Spark compatible SQL. It should not assume that source/original functions are hive specific. 
Can we rename avoidable references to hive and call them CoralIR instead?

Reply:
I changed it to be `FallBackToLinkedInHiveUDFTransformer`, keep using `HiveUDF` because it only falls back the Hive specific UDFs (starting with `com.linkedin.`).